### PR TITLE
PROD-175/mixpanel-spike 

### PR DESCRIPTION
**Pull request by danijel-radomiljac**
We are showing userID in every event.

In login we are showing wallet connector.

In question reveal we are showing transactionSignature, burnedAmount and questionId.

In claimed question we are showing transactionSignature, claimedAmount and questionId.

Screenshot 2024-08-28 at 13 10 26

### CODE REVIEW

**Comment by Yihwan** [comment]
also defer to @marvinmarnold for actual code review

File: package.json [hunk 1]
```diff
@@ -55,6 +55,8 @@
     "dayjs": "^1.11.10",
     "gsap": "^3.12.5",
     "jsonwebtoken": "^9.0.2",
+    "mixpanel": "^0.18.0",
+    "mixpanel-browser": "^2.55.0",
```
Comment (marvinmarnold) [inline package.json:59]: Is this actually being used? If not, let's remove.

**Review by marvinmarnold — CHANGES_REQUESTED** [review]

**Comment by danijel-radomiljac** [comment]
> thanks! some questions:
> 
> * is it possible to get city/country data with server side analytics? or is that only available client side?
> * can you send the actual question content in addition to question IDs? primary users of this dashboard will likely be non technical/not be able to lookup questions by ID easily. we might add more question-related content later, but for now i think just ID and the actual question text is sufficient.

yes we can track city/country but we would need to use some external service and probably one of paid versions: https://ipapi.com/product

File: lib/mixpanel.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  eventProperties?: Record<string, any>,
+) => {
+  const payload = await getJwtPayload();
+
+  if (!payload) return;
+
+  const userUUID = payload.sub;
+
+  const additionalProperties = {
+    distinct_id: userUUID,
```
Comment (marvinmarnold) [inline lib/mixpanel.ts:14]: @danijel-radomiljac it looks like if we include `ip`, Mixpanel will automatically geolocate

Screenshot 2024-08-29 at 10 34 20 AM

**Review by marvinmarnold — CHANGES_REQUESTED** [review]

File: app/api/mixpanel/route.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import Mixpanel from "mixpanel";
+
+const mixpanel = Mixpanel.init(process.env.MIX_PANEL_TOKEN!);
+
+export async function POST(request: Request) {
+  const data = await request.json();
+  try {
+    const { event, properties } = data;
+
+    mixpanel.track(event, properties);
```
Comment (marvinmarnold) [inline app/api/mixpanel/route.ts:12]: @danijel-radomiljac after second look, I actually think it's better to pull IP directly from the request on the server: https://nextjs.org/docs/pages/api-reference/functions/next-request#ip

That way we don't have to do an extra API request client side.

**Review by marvinmarnold — CHANGES_REQUESTED** [review]

**Review by marvinmarnold — APPROVED** [review]

**Pull request merged**